frontend/src/components/student/studentStatusButton.tsx

```diff
@@ -0,0 +1,37 @@
+import { Button } from "react-bootstrap";
+import { Status } from "../../api/entities/StudentEntity";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+export function StudentStatusButton(props: { status: Status; colour: string }) {
+    const [clicked, setClicked] = useState(false);
+    const router = useRouter();
+
+    useEffect(() => {
+        setClicked(router.query?.status === props.status);
+    }, [router.query]);
```

Comment:
To reduce re-renders, make a constant of query.status and be only dependent on that.